pyam/utils.py

```diff
@@ -221,10 +242,9 @@ def find_depth(data, s, level):
     else:
         raise ValueError('Unknown level type: {}'.format(level))
 
-    # determine depth
-    pipe = re.compile('\\|')
-    regexp = str(s).replace('*', '')
-    apply_test = lambda val: test(len(pipe.findall(val.replace(regexp, ''))))
+    def apply_test(val):
```

Comment:
suggest making this a lambda similarly:

```
list(map(lambda x: test(_find_depth(x)), data))
```

Reply:
@gidden, there is a plethora of tests for `level` in `test_core.py` - do you want me to refactor them and migrate to `test_utils.py`?

Reply:
done